sdk/inc/azure/core/az_result.h

```diff
@@ -133,26 +144,32 @@ typedef enum
       8), ///< Generic error in the HTTP transport adapter implementation.
 
   // IoT error codes
+
+  /// The IoT topic is not matching the expected format.
```

Comment:
Added in https://github.com/Azure/azure-sdk-for-c/pull/565. @danewalton - review the doc comment please.

Reply:
I pulled the comment from IoT's doxygen descriptions, which are very consistent when they are talking about this error code.
https://github.com/Azure/azure-sdk-for-c/search?q=%22+not+matching+the+expected+format%22&unscoped_q=%22+not+matching+the+expected+format%22